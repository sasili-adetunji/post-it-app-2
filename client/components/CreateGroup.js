import React from 'react';
import { CardHeader, CardTitle } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Paper from 'material-ui/Paper';
import PostItActions from '../actions/PostItActions';


const style = {
  height: 100,
  width: 100,
  margin: '30px 30px',
  padding: 30,
  textAlign: 'center',
  display: 'inline-block',
};

/**
 *
 *
 * @class CreateGroup
 * @extends {React.Component}
 */
class CreateGroup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      groupname: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }


  /**
   * monitors the state of the components and change the state
   *
   * @param {any} e
   * @memberof CreateGroup
   */

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  /**
   * makes an action call to create group
   *
   * @param {any} e
   * @memberof CreateGroup
   */
  onClick(e) {
    e.preventDefault();
    const group = {
      groupname: this.state.groupname
    };
    PostItActions.createGroup(group);
  }
  /**
   *
   * renders the createGroup components
   * @returns {void}
   * @memberof CreateGroup
   */
  render() {
    return (
      <div>
        <CardTitle title="Create" />
        <TextField
          name="groupname" onChange={this.onChange} value={this.state.groupname}
          floatingLabelText="Group Name" /><br />
        <RaisedButton
          style={{
            width: '200px',
            borderColor: '#D0D0D0',
            resize: 'none',
            borderRadius: 3,
            minHeight: '50px',
            fontSize: 14
          }} onClick={this.onClick}
              label="Create" primary />
      </div>
    );
  }

}
export default CreateGroup;
