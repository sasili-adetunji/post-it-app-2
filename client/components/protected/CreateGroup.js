import React from 'react';
import PostItActions from '../../actions/PostItActions';


/**
 * create creategroup components
 *
 * @class CreateGroup
 * @extends {React.Component}
 */
class CreateGroup extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      groupName: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }

  onChange(e) {
    this.setState({
      groupName: e.target.value
    });
  }
  onClick(e) {
    e.preventDefault();
    const group = {
      groupName: this.state.groupName,
      userName: this.props.userName.displayName
    };
    PostItActions.createGroup(group);
    this.setState({
      groupName: ''
    });
  }
  render() {
    return (
      <div className="panel-body">
        <form className="navbar-form" role="search">
          <div className="form-group">
            <input
            type="text" className="form-control" placeholder="Create Group"
            name="groupName" onChange={this.onChange} value={this.state.groupName} />
          </div>
          <button onClick={this.onClick} type="submit" className="btn btn-default ">
            <span className="glyphicon glyphicon-plus" /></button>
        </form>
      </div>
    );
  }
}
export default CreateGroup;
