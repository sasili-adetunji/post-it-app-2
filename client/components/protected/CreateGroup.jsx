import React from 'react';
import PostItActions from '../../actions/PostItActions';
import * as API from '../../Api';


/**
 * create creategroup components
 *
 * @class CreateGroup
 * 
 * @extends {React.Component}
 */
class CreateGroup extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      groupName: '',
      error: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }

  /**
  * @method onChange
  *
  * @description Monitors changes in the components and change the state
  *
  * @memberof CreateGroup
  *
  * @param {object}
  *
  * @returns {void}
  */

  onChange(event) {
    this.setState({
      groupName: event.target.value
    });
  }


/**
 * @description Makes an action call to create a group
 *
 * @param {object} event
 * 
 * @returns {void}
 * 
 * @memberof CreateGroup
*/
  onClick(event) {
    event.preventDefault();
    const group = {
      groupName: this.state.groupName,
    };
    if (!this.state.groupName) {
      this.setState({
        error: 'Group Name is Required'
      })
    } else {
    PostItActions.createGroup(group);
    API.getUserGroups();
    this.setState({
      groupName: '',
      error: ''
    });
    }
  }


  /**
   * @method render
   * 
   * Render react component
   * 
   * @returns {String} The HTML markup for the CreateGroup Components
   * 
   * @memberof CreateGroup
   */
  render() {
    return (
      <div className="panel-body">
        <div className='error'> {this.state.error} </div>
        <form className="navbar-form" role="search">
          <div className="form-group">
            <input
            type="text" className="form-control" placeholder="Create Group"
            name="groupName" onChange={this.onChange} 
            value={this.state.groupName} />
          </div>
          <button onClick={this.onClick} type="submit" 
          className="btn btn-default ">
            <span className="glyphicon glyphicon-plus" /></button>
        </form>
      </div>
    );
  }
}
export default CreateGroup;
