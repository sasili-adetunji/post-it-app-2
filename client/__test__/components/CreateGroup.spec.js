import React from 'react';
import { shallow, render } from 'enzyme';
import CreateGroup from '../../components/protected/CreateGroup.jsx';


function setup() {
  const props = {
    onClick: () => {},
    onChange: () => {},

  };
  return shallow(<CreateGroup {...props} />);
}

describe('CreateGroup components', () => {
  const component = setup();
  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain two div', () => {
    expect(component.find('div').length).toEqual(2);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});

describe('CreateGroup  Test', () => {
  it('should take props', () => {
    const component = setup();
    expect(component.props().onClick).toExist;
    expect(component.props().onChange).toExist;
  });
});
