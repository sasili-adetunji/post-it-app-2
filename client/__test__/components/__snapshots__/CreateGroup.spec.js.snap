// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroup components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="panel-body"
  >
    <form
      className="navbar-form"
      role="search"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          name="groupName"
          onChange={[Function]}
          placeholder="Create Group"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-default "
        onClick={[Function]}
        type="submit"
      >
        <span
          className="glyphicon glyphicon-plus"
        />
      </button>
    </form>
  </div>,
  "nodes": Array [
    <div
      className="panel-body"
    >
      <form
        className="navbar-form"
        role="search"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="groupName"
            onChange={[Function]}
            placeholder="Create Group"
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-default "
          onClick={[Function]}
          type="submit"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </button>
      </form>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <CreateGroup
      onChange={[Function]}
      onClick={[Function]}
    />,
    "_instance": CreateGroup {
      "context": Object {},
      "onChange": [Function],
      "onClick": [Function],
      "props": Object {
        "onChange": [Function],
        "onClick": [Function],
      },
      "refs": Object {},
      "state": Object {
        "groupName": "",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="panel-body"
    >
      <form
        className="navbar-form"
        role="search"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="groupName"
            onChange={[Function]}
            placeholder="Create Group"
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-default "
          onClick={[Function]}
          type="submit"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </button>
      </form>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CreateGroup
    onChange={[Function]}
    onClick={[Function]}
  />,
}
`;
